storage/rocksdb/ha_rocksdb.cc

```diff
@@ -8317,6 +8368,9 @@ int ha_rocksdb::update_indexes(const struct update_row_info &row_info,
                                const bool &pk_changed) {
   int rc;
 
+  // If this table has a non-hidden PK, verify it will be updated first for TTL
+  DBUG_ASSERT_IMP(!has_hidden_pk(table), table->s->primary_key == 0);
```

Comment:
while I think this *should* always be true, I still think it's better to explicitly update the pk first even if it makes the code a little less clean.  `DBUG_ASSERT_IMP` are ignored in production so if in some weird scenario it's not true it's going to cause problems (it'll put the wrong ttl into secondary keys? so I don't think it's worth risking data corruption).